### ISSUE TITLE
Cannot provide a preloaded `resources` object in configuration and lazy-load other languages

### ISSUE THREAD

**Issue by scottrabin** [i18next/i18next#617]
When using the [XHR backend](https://github.com/i18next/i18next-xhr-backend), all languages lazy-load without issue... as long as I don't add a `resources` object to the initialization hash. As an example:

```js
i18next.use(XHR).init({
    backend: {
        loadPath: 'correct/path/to/{{lng}}.json',
    },
    fallbackLng: 'en-US',
});

// elsewhere...
i18next.changeLanguage('de-DE', function(err, t) {
    // `t` correctly translates messages into the specified language
});
```

The above will cause the module to load the `en-US` language bundle immediately. If I try to provide it with the language translations via seeding the initial resources bundle, however, languages will stop lazy loading entirely.

```js
i18next.use(XHR).init({
    backend: {
        loadPath: 'correct/path/to/{{lng}}.json',
    },
    fallbackLng: 'en-US',
    resources: {
        "en-US": {
             "translation": require('relative/path/to/en-US.json'),
        },
    },
});

// elsewhere...
i18next.changeLanguage('de-DE', function(err, t) {
    // `t` is still en-US translation function
});
```

I believe [this line](https://github.com/i18next/i18next/blob/master/src/i18next.js#L119) to be at fault. It seems that it would be easy to fix this issue by simply omitting this check (whether or not the user pre-seeded the resources with a given object) entirely; however, as I do not understand the historical context around this line, I am not sure what might break as a result of removing the predicate. Short-circuiting the predicate does seem to fix the lazy loading issue, while still using the seeded value for the default language.

Is this simply a remnant of previous implementations before an explicit caching layer was introduced? Or would many downstream consumers be affected by suddenly opening them up to making requests via a backend and/or using a caching layer, where these were simply bypassed before?

**Comment by jamuhl** [comment]
if you pass in resources on init - we expect you to not use a backend to load stuff but instead 'hardcoded' that into it.

so if you need both, don't add resources on init but instead use the addResourceBundle functions: http://i18next.com/docs/api/#add-resource-bundle

**Comment by scottrabin** [comment]
Even if I call `addResourceBundle` immediately after the `init` call, it still attempts to load `en-US` via XHR. Any way to bypass that?

**Comment by jamuhl** [comment]
hm...the only option i see right now is to set lng='cimode' on init - and after addResourceBundle call changeLanguage to users language. But that's more a hack.

A better way would be to introduce some option `startOnInit` which defaults to true. Setting it to false will stop execution of this line: https://github.com/i18next/i18next/blob/master/src/i18next.js#L106. Further we add a new function `start` on i18next which triggers the changeLanguage call with this.options.language

what do you think?

**Comment by scottrabin** [comment]
I tend to avoid introducing additional configuration/API bloat; it adds unnecessary complexity and can be confusing as to behaviors in the presence/absence of certain configuration options. Seeing as the languages load asynchronously anyway, why not put the `changeLanguage` call inside a `setImmediate`, so that the current execution stack can complete before it even attempts to initiate the fetch? This avoids any changes to the API and maintains current functionality, as `changeLanguage` is already asynchronous. I could call `init` followed immediately by `addResourceBundle`, and i18next would see the preloaded bundle before initiating the XHR - which gives me the behavior I want (a preloaded, network-free language bundle).

Out of curiosity - if the keys of `whitelist` and the provided bundles in `resources` don't match, what would you think it should do? To me, it seems that bypassing the XHR step for missing keys when the XHR backend is being `use`d is unexpected.

**Comment by jamuhl** [comment]
setImmediate is not standard. not sure but doing a setTimeout with some milliseconds might work too?

whitelist...i think it would not allow this language at all if not set...even if manually added to resourceStore

**Comment by jamuhl** [comment]
just published v2.4.0 which calls loading inside a timeout block...this should enable you to add the resourceBundle upfront (i hope)

**Comment by scottrabin** [comment]
Yep, works exactly as expected by simply appending `i18next.addResourceBundle('en-US', 'translation', require('path/to/en-US.json'));`! Thanks for the quick turnaround!
